### PR TITLE
FEATURE - Add option to use a local config file instead of the default config.json

### PR DESCRIPTION
**Pull request by NL-TCH**
The install.sh script creates its own config file, and there is no way to edit this before installing/starting. This function -l or --local-config accepts your json file.

**example without config file**

```text
./install.sh

          ▄▄▄▄  ▄▄▄▄▄  ▄▄   ▄
         █▀   ▀ █   ▀█ █▀▄  █
         ▀█▄▄▄  █▄▄▄█▀ █ █▄ █
             ▀█ █      █  █ █
         ▀▄▄▄█▀ █      █   ██
        Safing Privacy Network

-> Please enter metrics comment:
> 
-> Installation configuration:

   Architecture: amd64
   Download-URL: https://updates.safing.io/latest/linux_amd64/start/portmaster-start
   Target Dir: /opt/safing/spn
   Install systemd: yes
   Unit path: /etc/systemd/system/spn.service
   Start Now: yes
   Config File: default

   Config:
{
"core": {
    "metrics": {
    "instance": "virtual-machine",
    "comment": "",
    "push": ""
    }
},
"spn": {
    "publicHub": {
    "name": "virtual-machine"
    }
}
}


Continue (Y/n)? n
```

**example with config file**
```text
./install.sh -l config.json 

          ▄▄▄▄  ▄▄▄▄▄  ▄▄   ▄
         █▀   ▀ █   ▀█ █▀▄  █
         ▀█▄▄▄  █▄▄▄█▀ █ █▄ █
             ▀█ █      █  █ █
         ▀▄▄▄█▀ █      █   ██
        Safing Privacy Network

-> Please enter metrics comment:
> 
-> Installation configuration:

   Architecture: amd64
   Download-URL: https://updates.safing.io/latest/linux_amd64/start/portmaster-start
   Target Dir: /opt/safing/spn
   Install systemd: yes
   Unit path: /etc/systemd/system/spn.service
   Start Now: yes
   Config File: config.json

   Config:
{
    "core": {
      "metrics": {
        "instance": "justsomename",
        "comment": "docker instance build by NL-TCH",
        "push": ""
      }
    },
    "spn": {
      "publicHub": {
        "name": "justsomename",
        "ip4": "123.12.1.123"
      }
    }
  }

Continue (Y/n)? n
```

**the config.json**
```json
{
    "core": {
      "metrics": {
        "instance": "justsomename",
        "comment": "docker instance build by NL-TCH",
        "push": ""
      }
    },
    "spn": {
      "publicHub": {
        "name": "justsomename",
        "ip4": "123.12.1.123"
      }
    }
  }
```

### CODE REVIEW

File: tools/install.sh [hunk 1]
```diff
@@ -188,35 +189,42 @@ ask_config() {
...
-  "spn": {
+},
+"spn": {
     "publicHub": {
-      "name": "$HOSTNAME"
+    "name": "$HOSTNAME"
     }
-  }
+}
 }
```
Comment (dhaavi) [inline tools/install.sh:209]: Please don't change the indentation here.

File: tools/install.sh [hunk 2]
```diff
@@ -188,35 +189,42 @@ ask_config() {
...
 
 write_config_file() {
-    cat >${1} <<EOT
+    local config_file=${1:-${LOCAL_CONFIG_FILE:-default}}
```
Comment (dhaavi) [inline tools/install.sh:192]: I'm not sure what this is for. Why is the fallback value "default"? It also seems that `config_file` is not used anyway.

**Review by dhaavi — CHANGES_REQUESTED** [review]
Hey, thanks for this contribution. I think having the option to specify a prepared config file is a nice addition.

If I understand correctly, when a config file is specified, the config file should not be written. This also means, that we should not ask for any configuration in the ask_config() function.

I also added some comments to the PR.

File: tools/install.sh (hunk 1 above)
Comment (NL-TCH) [reply tools/install.sh]: thanks for the feedback, see the new commit

File: tools/install.sh (hunk 2 above)
Comment (NL-TCH) [reply tools/install.sh]: thanks for the feedback, see the new commit

**Pull request closed without merging**